metaflow/plugins/kubernetes/kubernetes_client.py

```diff
@@ -32,15 +34,19 @@ def __init__(self):
     def _refresh_client(self):
         from kubernetes import client, config
 
-        if os.getenv("KUBERNETES_SERVICE_HOST"):
-            # We are inside a pod, authenticate via ServiceAccount assigned to us
-            config.load_incluster_config()
-        else:
+        if os.getenv("KUBECONFIG"):
             # Use kubeconfig, likely $HOME/.kube/config
             # TODO (savin):
             #  1. Support generating kubeconfig on the fly using boto3
             #  2. Support auth via OIDC - https://docs.aws.amazon.com/eks/latest/userguide/authenticate-oidc-identity-provider.html
             config.load_kube_config()
+        elif os.getenv("KUBERNETES_SERVICE_HOST"):
+            # We are inside a pod, authenticate via ServiceAccount assigned to us
+            config.load_incluster_config()
+        else:
```

Comment:
this might be a backwards incompatible change and will break many currently well functioning flows. do we expect `KUBECONFIG` env var to be always available?

Reply:
This could be made backward compatible if the original call to `config.load_kube_config` in the `else` case is retained instead of throwing the KubernetesException.